### ISSUE TITLE
In non-Evil use, `vimish-fold` conflicts with `AUCTeX` and disables `'TeX-mode-map` keybindings beginning with `C-c C-f` 

### ISSUE THREAD

**Issue by kjhealy** [hlissner/doom-emacs#5951]
### What did you expect to happen?

I use Doom Emacs but not with Evil keybindings (i.e. in my `init.el`, `(evil +everywhere)` is disabled). 

When editing a LaTeX file with AUCTeX's `latex-mode`, I expect to be able to type any of several key sequences  beginning `C-c C-f` and have some LaTeX formatting command inserted. For example, `C-c C-f C-e` should produce `\emph{}`, or `C-c C-f C-b` should produce `\textbf{}`. 

These key sequences are created by `TeX-mode-map`, which AUCTeX binds to `C-c C-f`. 

### What actually happened?

What actually happens is that, even though AUCTeX is loaded and the relevant commands and keyboard shortcuts can be seen and are listed in the menu under LaTeX > Insert Font, these keybindings do not work. Instead, Emacs reports, e.g.,

`C-c C-f C-e is undefined`

Other AUCTeX keybindings (e.g. `C-c C-s` for "New Section" and so on) continue to work as usual.

### Describe your attempts to resolve the issue

This issue appears to be caused by the `vimish-fold` package somehow disabling these particular keybindings in AUCTeX. Removing `vimish-fold` from `init.el`, by commenting out the line 

```
;;fold
```
in the `:editor` section and then and re-syncing with `doom sync` solves the issue. It seems that `vimish-fold`  defines a menu sequence that is triggered by `C-c C-f` (so that e.g. `C-c C-f C-d` triggers `vimish-fold-delete`). In the process it unbinds all the formatting bindings that AUCTeX's `TeX-mode-map` assigns under `C-c C-f`.

### Steps to reproduce

1. In `init.el`, ensure that `;;(evil +everywhere)` is _disabled_. 
2. In `init.el`, ensure that `fold` is _enabled_.  
3. In `init.el`, ensure that `latex` is _enabled_. 
4. Open a LaTeX file. 
5. Try to insert, e.g. `C-c C-f C-e` for `\emph{}`.
6. Modeline error message: `C-c C-f C-e is undefined`.

### System Information

https://pastebin.com/HtNCKfht

**Comment by vancleve** [comment]
omg, yes, just ran into this!

**Comment by vancleve** [comment]
Here is where the keybindings for this are set:
https://github.com/hlissner/doom-emacs/blob/42e5763782fdc1aabb9f2624d468248d6978abe2/modules/config/default/%2Bemacs-bindings.el#L635

AUCTeX has its own fold facilities it seems, 
<https://www.gnu.org/software/auctex/manual/auctex/Folding.html>
with prefix `C-o`. 

Options could be:
1. not enable `fold` keybindings in when AUCTeX is enabled (though I don't know how to do that). Then one could rely on AUCTeX's fold mechanism.
2. give `fold` a global prefix of `C-o` which I don't think would overlap much with other things...?

Personally, I'd go for option 1 but I don't use folding much in latex. However, I do use the font commands, so its be great to have them back with the `C-c C-f` prefix.

**Comment by vancleve** [comment]
Coming back to this, here is how I fixed this in my `config.el`:

```
(after! tex
  (map! :leader
      (:when (modulep! :editor fold)
       ("C-f" nil))))

**Comment by liar666** [comment]
1000s thanks for the workaround!! I was really in pain having to use the mouse & menus just for adding some emphasis :)

**Comment by LemonBreezes** [comment]
Yeah, I believe that the non-evil Fold keybindings are not good. `C-c C-f` is not a user prefix. I have this code in my non-Evil config to use `C-c F` instead
```elisp
;; Doom binds it's folding prefix to `C-c C-f' which is a keybinding used by
;; many major modes.
(when (modulep! :editor fold)
  (defvar doom-fold-map (lookup-key doom-leader-map "\C-f"))
  (define-key doom-leader-map "\C-f" nil)
  (map! :leader
        (:prefix-map ("F" . "fold")
         "k" #'vimish-fold-delete
         "K" #'vimish-fold-delete-all
         "c" #'vimish-fold
         "t" #'+fold/toggle
         "C" #'+fold/close-all
         "o" #'+fold/open
         "O" #'+fold/open-all)))
```
